hil/src/commands/ota.rs

```diff
@@ -0,0 +1,634 @@
+        }
+
+        info!("Starting update progress and service status monitoring");
+        if let Err(e) = self.monitor_update_progress(&session).await {
+            println!("OTA_RESULT=FAILED");
+            println!("OTA_ERROR=UPDATE_PROGRESS_FAILED: {e}");
+            return Err(e);
+        }
+
+        let session = match self.handle_reboot("update").await {
+            Ok(session) => {
+                info!("Device successfully rebooted and reconnected - update application completed");
+                session
+            }
+            Err(e) => {
+                println!("OTA_RESULT=FAILED");
+                println!("OTA_ERROR=POST_UPDATE_REBOOT_FAILED: {e}");
+                return Err(e);
+            }
+        };
+
+        if let Err(e) = self.run_update_verifier(&session).await {
+            println!("OTA_RESULT=FAILED");
+            println!("OTA_ERROR=UPDATE_VERIFIER_FAILED: {e}");
+            return Err(e);
+        }
+
+        if let Err(e) = self.run_check_my_orb(&session).await {
+            println!("OTA_RESULT=FAILED");
+            println!("OTA_ERROR=CHECK_MY_ORB_FAILED: {e}");
+            return Err(e);
+        }
+
+        if let Err(e) = self.get_boot_time(&session).await {
+            println!("GET_BOOT_TIME=FAILED: {e}");
+        }
+
+        println!("OTA_RESULT=SUCCESS");
+        println!("OTA_VERSION={}", self.version);
+        println!("OTA_SLOT_FINAL={}", current_slot);
+        println!("OTA_WIPE_OVERLAYS_FINAL={}", wipe_overlays_status);
+
+        info!("OTA update completed successfully!");
+        Ok(())
+    }
+
+    async fn connect(&self) -> Result<SshWrapper> {
+        info!("Connecting to Orb device at {}:{}", self.host, self.port);
+
+        let session = SshWrapper::connect(
+            self.host.clone(),
+            self.port,
+            self.username.clone(),
+            self.password.clone(),
+        )
+        .await
+        .wrap_err("Failed to establish SSH connection to Orb device")?;
+
+        info!("Successfully connected to Orb device");
+        Ok(session)
+    }
+
+    #[instrument(skip(self, session))]
+    async fn get_current_slot(&self, session: &SshWrapper) -> Result<String> {
+        info!("Determining current slot");
+        let result = session
+            .execute_command("TERM=dumb orb-slot-ctrl -c")
+            .await
+            .wrap_err("Failed to execute orb-slot-ctrl -c")?;
+
+        if !result.is_success() {
+            bail!("orb-slot-ctrl -c failed: {}", result.stderr);
+        }
+
+        let slot_regex = Regex::new(r"(a|b)")?;
+
+        if let Some(captures) = slot_regex.captures(&result.stdout) {
+            let slot_letter = captures.get(1).unwrap().as_str();
+            let slot_name = format!("slot_{slot_letter}");
+            info!("Current slot: {}", slot_name);
+            Ok(slot_name)
+        } else {
+            bail!("Could not parse current slot from: {}", result.stdout);
+        }
+    }
```

Comment:
nit: This and a lot of the methods could be a free function rather than an associated function (aka method). Because it doesn't use much at all from `self`.

I find that it is more verbose but more explicit and less OOP to try to use free-standing functions when possible.

Good rule of thumb - if you start to need more than two of the fields of `self`, then `self` is ok to use (or consider breaking the fields up into logical containers, for example all of your ssh dialing info (user, password, port, address) could be a SshDialInfo struct.